content/articles/how-to-lookup-ip-addresses-in-java/index.md

```diff
@@ -0,0 +1,225 @@
+
+#### Java InetAddress Class Methods
+If necessary, the three methods may connect to a local DNS server to fill up the information in the InetAddress objects. Always remember that these methods make network connections to retrieve all information they need.
+
+The InetAddress class caches the results of the DNS lookups such that once it has the address of the given host, it won't have to look it up again. If the IP address does not change while your program is running, there will be no issues.
+
+Negative results such as `host not found` are slightly problematic. It happens for the first time, and in the second trial, one succeeds. This error will occur when the first attempt timed out while the information was in transit from the remote DNS server.
+
+#### public static InetAddress getByName(String hostName) throws UnknownHostException
+`InetAddress.getByName()` is the frequently used method. It is a static method that takes the hostname that you are looking for as its arguments. It looks up the host's IP address using DNS. 
+
+It is called: `getByName()`. Assuming there is an import **java.net.\*** statement at the top of the program and any other necessary import statements.
+
+```java
+ InetAddress address = InetAddress.getByName("www.google.com");
+```
+
+#### Exception handling mechanism used
+The `InetAddress.getByName()` method throws UnknownHostException if host can't be found.
+
+It can either be declared using the `throws` exception or wrapped in a `try-catch` block. As shown below:
+
+```java
+try{
+    InetAddress address = InetAddress.getByName("www.google.com");
+    System.out.println(address);
+}
+catch(UnknownHostException ex){
+    System.out.println("www.google.com not found");
+}
+
+```
+
+### Commonly used methods.
+Below is a sample code using `getByName()` method:
+
+```java
+package com. company;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+public class GetIpAddress {
+   public static void main(String[]args) throws UnknownHostException{
+        InetAddress address = InetAddress.getByName("www.microsoft.com");
+
+        System.out.println("my local machine: " + address);
+    }
+}
+```
+
+A sample output of this code:
+
+```bash
+my local machine: www.microsoft.com/2.21.100.214
+
+```
+
+> For machines that do not have hostnames, you can always pass a string that contains a hexadecimal form of the IP address to `InetAddress.getByName()`.
+
+Example:
+
+```java
+InetAddress address = InetAddress.getByName("172.217.170.164");
+
+```
+
+> Hostnames are more stable than IP addresses. Most services have lived at the same hostnames but have changed their IP addresses severally. When choosing between hostnames like `www.google.com` and IP addresses like `172.217.170.164`, always go with the hostname.
+
+#### public static InetAddress[]getAllByName(String hostName) throws UnknownHostException

```

Comment:
This should not be a subheading. If it is a note change it to the appropriate method to represent things the reader should note. You can use `>` to identify a note. Do the same for the rest.